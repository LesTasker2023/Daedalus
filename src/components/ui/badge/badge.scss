// TriggerKings Badge Component - Pure SCSS Implementation
// Enhanced visual design with more prominent styling

.badge {
  // Base Badge Styles - More Prominent Visual Foundation
  display: inline-flex;
  align-items: center;
  justify-content: center;
  border-radius: 0.5rem; /* 8px - slightly more rounded */
  padding: 0.25rem 0.75rem; /* 4px 12px - more padding */
  font-size: 0.75rem; /* 12px */
  font-weight: 600; /* semibold for better readability */
  line-height: 1.2;
  white-space: nowrap;
  border: 1px solid transparent;
  transition: all 0.15s ease-in-out;
  box-shadow: 0 1px 2px 0 rgb(0 0 0 / 0.05); /* subtle shadow */

  // Focus Management - Enhanced Accessibility
  &:focus-visible {
    outline: 2px solid hsl(var(--ring));
    outline-offset: 2px;
    box-shadow: 0 0 0 2px hsl(var(--ring) / 0.2);
  }

  // SVG Icons Support - Consistent Sizing
  svg {
    width: 0.75rem; /* 12px */
    height: 0.75rem; /* 12px */
    pointer-events: none;
  }

  // Gap for content with icons
  &:has(svg) {
    gap: 0.25rem; /* 4px */
  }

  // Variant: Default (Primary Brand) - Enhanced
  &--default {
    background-color: hsl(var(--primary));
    color: hsl(var(--primary-foreground));
    border-color: transparent;
    box-shadow: 0 1px 3px 0 hsl(var(--primary) / 0.3);

    // Interactive States
    &:hover {
      background-color: hsl(var(--primary) / 0.85);
      box-shadow: 0 2px 4px 0 hsl(var(--primary) / 0.4);
      transform: translateY(-1px);
    }

    // Active/Pressed State
    &:active {
      background-color: hsl(var(--primary) / 0.9);
      transform: translateY(0);
      box-shadow: 0 1px 2px 0 hsl(var(--primary) / 0.3);
    }
  }

  // Variant: Secondary (Muted) - Enhanced
  &--secondary {
    background-color: hsl(var(--secondary));
    color: hsl(var(--secondary-foreground));
    border-color: hsl(var(--border) / 0.3);
    box-shadow: 0 1px 2px 0 rgb(0 0 0 / 0.05);

    // Interactive States
    &:hover {
      background-color: hsl(var(--secondary) / 0.85);
      border-color: hsl(var(--border) / 0.5);
      box-shadow: 0 2px 4px 0 rgb(0 0 0 / 0.1);
      transform: translateY(-1px);
    }

    // Active/Pressed State
    &:active {
      background-color: hsl(var(--secondary) / 0.9);
      transform: translateY(0);
    }
  }

  // Variant: Destructive (Error/Warning) - Enhanced
  &--destructive {
    background-color: hsl(var(--destructive));
    color: hsl(var(--destructive-foreground));
    border-color: transparent;
    box-shadow: 0 1px 3px 0 hsl(var(--destructive) / 0.3);

    // Interactive States
    &:hover {
      background-color: hsl(var(--destructive) / 0.85);
      box-shadow: 0 2px 4px 0 hsl(var(--destructive) / 0.4);
      transform: translateY(-1px);
    }

    // Active/Pressed State
    &:active {
      background-color: hsl(var(--destructive) / 0.9);
      transform: translateY(0);
    }

    // Enhanced focus for destructive actions
    &:focus-visible {
      box-shadow:
        0 0 0 2px hsl(var(--background)),
        0 0 0 4px hsl(var(--destructive) / 0.5);
    }
  }

  // Variant: Outline (Subtle) - Enhanced
  &--outline {
    background-color: transparent;
    color: hsl(var(--foreground));
    border-color: hsl(var(--border));
    border-width: 1.5px; /* slightly thicker border */
    box-shadow: 0 1px 2px 0 rgb(0 0 0 / 0.05);

    // Interactive States
    &:hover {
      background-color: hsl(var(--accent));
      color: hsl(var(--accent-foreground));
      border-color: hsl(var(--border) / 0.8);
      box-shadow: 0 2px 4px 0 rgb(0 0 0 / 0.1);
      transform: translateY(-1px);
    }

    // Active/Pressed State
    &:active {
      background-color: hsl(var(--accent) / 0.9);
      transform: translateY(0);
    }
  }

  // Badge as Link/Button Enhancement - Applied when used with asChild
  &:hover:is(a, button) {
    cursor: pointer;
    text-decoration: none;
  }

  &:focus-visible:is(a, button) {
    outline: 2px solid hsl(var(--ring));
    outline-offset: 2px;
  }

  // Badge with Number/Count Support - Enhanced
  &[data-count] {
    min-width: 1.5rem; /* 24px - slightly larger */
    height: 1.5rem; /* 24px */
    border-radius: 50%;
    padding: 0 0.5rem; /* 8px - more padding */
    font-family: var(--font-mono, monospace);
    font-weight: 700; /* bold for numbers */
    font-feature-settings: "tnum";
    font-variant-numeric: tabular-nums;
    box-shadow: 0 2px 4px 0 rgb(0 0 0 / 0.1);

    // Enhanced shadow for count badges
    &.badge--default {
      box-shadow: 0 2px 6px 0 hsl(var(--primary) / 0.4);
    }

    &.badge--destructive {
      box-shadow: 0 2px 6px 0 hsl(var(--destructive) / 0.4);
    }
  }

  // Dark Mode Optimizations
  @media (prefers-color-scheme: dark) {
    // Future dark mode enhancements
  }

  // High Contrast Mode Support
  @media (prefers-contrast: high) {
    border-width: 2px;

    &--outline {
      border-color: currentColor;
    }
  }

  // Reduced Motion Support
  @media (prefers-reduced-motion: reduce) {
    transition: none;

    &:active {
      transform: none;
    }
  }

  // Print Styles
  @media print {
    background: none !important;
    color: black !important;
    border: 1px solid black !important;
  }
}
