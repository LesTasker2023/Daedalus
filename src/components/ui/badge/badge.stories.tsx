import type { Meta, StoryObj } from '@storybook/nextjs';
import Link from 'next/link';
import { Badge } from './badge';

// Icons for examples (using simple SVGs)
const CheckIcon = () => (
  <svg
    width="12"
    height="12"
    viewBox="0 0 24 24"
    fill="none"
    stroke="currentColor"
    strokeWidth="2"
    strokeLinecap="round"
    strokeLinejoin="round"
  >
    <polyline points="20,6 9,17 4,12" />
  </svg>
);

const AlertIcon = () => (
  <svg
    width="12"
    height="12"
    viewBox="0 0 24 24"
    fill="none"
    stroke="currentColor"
    strokeWidth="2"
    strokeLinecap="round"
    strokeLinejoin="round"
  >
    <circle cx="12" cy="12" r="10" />
    <line x1="12" y1="8" x2="12" y2="12" />
    <line x1="12" y1="16" x2="12.01" y2="16" />
  </svg>
);

const meta: Meta<typeof Badge> = {
  title: 'TriggerKings/Badge',
  component: Badge,
  parameters: {
    layout: 'centered',
    docs: {
      description: {
        component:
          'A versatile badge component for displaying status, counts, or categories. Built with enterprise-grade accessibility and design standards.',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    variant: {
      control: 'select',
      options: ['default', 'secondary', 'destructive', 'outline'],
      description: 'The visual variant of the badge',
    },
    asChild: {
      control: 'boolean',
      description: 'Render as child component (e.g., link)',
    },
    children: {
      control: 'text',
      description: 'Badge content',
    },
  },
  args: {
    children: 'Badge',
    variant: 'default',
    asChild: false,
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

// Basic Examples
export const Default: Story = {
  args: {
    children: 'Badge',
  },
};

export const Secondary: Story = {
  args: {
    children: 'Secondary',
    variant: 'secondary',
  },
};

export const Destructive: Story = {
  args: {
    children: 'Destructive',
    variant: 'destructive',
  },
};

export const Outline: Story = {
  args: {
    children: 'Outline',
    variant: 'outline',
  },
};

// Badge with Icons
export const WithIcon: Story = {
  render: () => (
    <div className="flex gap-2">
      <Badge variant="default">
        <CheckIcon />
        Verified
      </Badge>
      <Badge variant="destructive">
        <AlertIcon />
        Error
      </Badge>
    </div>
  ),
  parameters: {
    docs: {
      description: {
        story: 'Badges can include icons to provide additional visual context.',
      },
    },
  },
};

// Number/Count Badges
export const NumberBadges: Story = {
  render: () => (
    <div className="flex gap-2">
      <Badge data-count="true">5</Badge>
      <Badge variant="destructive" data-count="true">
        99+
      </Badge>
      <Badge variant="outline" data-count="true">
        12
      </Badge>
    </div>
  ),
  parameters: {
    docs: {
      description: {
        story: 'Circular badges perfect for counts, notifications, or numbers.',
      },
    },
  },
};

// Link Badges (asChild)
export const LinkBadges: Story = {
  render: () => (
    <div className="flex gap-2">
      <Badge asChild>
        <Link href="/">Home</Link>
      </Badge>
      <Badge variant="secondary" asChild>
        <Link href="/about">About</Link>
      </Badge>
      <Badge variant="outline" asChild>
        <a href="https://example.com" target="_blank" rel="noopener noreferrer">
          External
        </a>
      </Badge>
    </div>
  ),
  parameters: {
    docs: {
      description: {
        story: 'Badges can be rendered as links using the asChild prop.',
      },
    },
  },
};

// All Variants Showcase
export const AllVariants: Story = {
  render: () => (
    <div className="flex flex-wrap gap-2">
      <Badge>Default</Badge>
      <Badge variant="secondary">Secondary</Badge>
      <Badge variant="destructive">Destructive</Badge>
      <Badge variant="outline">Outline</Badge>
    </div>
  ),
  parameters: {
    docs: {
      description: {
        story: 'All available badge variants in one view.',
      },
    },
  },
};

// Status Badges
export const StatusBadges: Story = {
  render: () => (
    <div className="flex flex-wrap gap-2">
      <Badge variant="default">
        <CheckIcon />
        Active
      </Badge>
      <Badge variant="secondary">Pending</Badge>
      <Badge variant="destructive">
        <AlertIcon />
        Failed
      </Badge>
      <Badge variant="outline">Draft</Badge>
    </div>
  ),
  parameters: {
    docs: {
      description: {
        story: 'Common status indicators using different badge variants.',
      },
    },
  },
};

// Accessibility Example
export const AccessibilityExample: Story = {
  render: () => (
    <div className="flex gap-2">
      <Badge aria-label="5 unread messages" data-count="true">
        5
      </Badge>
      <Badge variant="destructive" role="alert" aria-label="Critical error occurred">
        <AlertIcon />
        Critical
      </Badge>
      <Badge variant="default" aria-describedby="badge-description">
        <CheckIcon />
        Verified
      </Badge>
      <div id="badge-description" className="sr-only">
        This item has been verified by our team
      </div>
    </div>
  ),
  parameters: {
    docs: {
      description: {
        story: 'Badges with proper ARIA labels and semantic meaning for screen readers.',
      },
    },
  },
};

// Custom Styling Example
export const CustomStyling: Story = {
  render: () => (
    <div className="flex gap-2">
      <Badge className="bg-blue-500 text-white hover:bg-blue-600">Custom Blue</Badge>
      <Badge className="bg-green-500 text-white hover:bg-green-600">Custom Green</Badge>
      <Badge className="bg-purple-500 text-white hover:bg-purple-600 rounded-full px-3 py-1">
        Custom Rounded
      </Badge>
    </div>
  ),
  parameters: {
    docs: {
      description: {
        story: 'Badges with custom styling using CSS classes.',
      },
    },
  },
};

// Interactive States
export const InteractiveStates: Story = {
  render: () => (
    <div className="flex gap-2">
      <Badge asChild>
        <button type="button">Clickable Badge</button>
      </Badge>
      <Badge variant="outline" asChild>
        <Link href="/profile">Profile Link</Link>
      </Badge>
      <Badge variant="destructive" asChild>
        <button type="button" onClick={() => alert('Delete action')}>
          Delete
        </button>
      </Badge>
    </div>
  ),
  parameters: {
    docs: {
      description: {
        story: 'Interactive badges as buttons and links with proper focus states.',
      },
    },
  },
};
