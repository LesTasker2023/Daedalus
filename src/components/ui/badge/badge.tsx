// TriggerKings Badge Component - Production Ready (shadcn API + Pure SCSS)
import { cn } from '@/lib/utils';
import { Slot } from '@radix-ui/react-slot';
import { cva, type VariantProps } from 'class-variance-authority';
import * as React from 'react';
import './badge.scss';

const badgeVariants = cva(
  'badge', // Base class
  {
    variants: {
      variant: {
        default: 'badge--default',
        secondary: 'badge--secondary',
        destructive: 'badge--destructive',
        outline: 'badge--outline',
      },
    },
    defaultVariants: {
      variant: 'default',
    },
  }
);

export interface BadgeProps
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof badgeVariants> {
  asChild?: boolean;
}

const Badge = React.forwardRef<HTMLDivElement, BadgeProps>(
  ({ className, variant, asChild = false, ...props }, ref) => {
    const Comp = asChild ? Slot : 'div';
    return <Comp className={cn(badgeVariants({ variant, className }))} ref={ref} {...props} />;
  }
);
Badge.displayName = 'Badge';

export { Badge, badgeVariants };
