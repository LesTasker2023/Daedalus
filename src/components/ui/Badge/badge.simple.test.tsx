import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { Badge } from './badge';

describe('Badge', () => {
  it('renders children correctly', () => {
    render(<Badge>Test Badge</Badge>);
    expect(screen.getByText('Test Badge')).toBeInTheDocument();
  });

  it('applies variant classes correctly', () => {
    render(<Badge variant="secondary">Secondary Badge</Badge>);
    const badge = screen.getByText('Secondary Badge');
    expect(badge).toHaveClass('bg-secondary');
  });

  it('supports custom className', () => {
    render(<Badge className="custom-class">Custom Badge</Badge>);
    const badge = screen.getByText('Custom Badge');
    expect(badge).toHaveClass('custom-class');
  });
});
