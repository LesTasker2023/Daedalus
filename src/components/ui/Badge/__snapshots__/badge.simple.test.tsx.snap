// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > Snapshots > matches all sizes snapshot 1`] = `
<div
  data-testid="sizes-container"
>
  <div
    class="inline-flex items-center rounded-md border font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-primary text-primary-foreground shadow hover:bg-primary/80 px-2 py-0.5 text-xs"
  >
    Small
  </div>
  <div
    class="inline-flex items-center rounded-md border font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-primary text-primary-foreground shadow hover:bg-primary/80 px-2.5 py-0.5 text-xs"
  >
    Default
  </div>
  <div
    class="inline-flex items-center rounded-md border font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-primary text-primary-foreground shadow hover:bg-primary/80 px-3 py-1 text-sm"
  >
    Large
  </div>
</div>
`;

exports[`Badge > Snapshots > matches all variants snapshot 1`] = `
<div
  data-testid="variants-container"
>
  <div
    class="inline-flex items-center rounded-md border font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-primary text-primary-foreground shadow hover:bg-primary/80 px-2.5 py-0.5 text-xs"
  >
    Default
  </div>
  <div
    class="inline-flex items-center rounded-md border font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-destructive text-destructive-foreground shadow hover:bg-destructive/80 px-2.5 py-0.5 text-xs"
  >
    Destructive
  </div>
  <div
    class="inline-flex items-center rounded-md border font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-foreground px-2.5 py-0.5 text-xs"
  >
    Outline
  </div>
  <div
    class="inline-flex items-center rounded-md border font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80 px-2.5 py-0.5 text-xs"
  >
    Secondary
  </div>
</div>
`;

exports[`Badge > Snapshots > matches default snapshot 1`] = `
<div
  class="inline-flex items-center rounded-md border font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-primary text-primary-foreground shadow hover:bg-primary/80 px-2.5 py-0.5 text-xs"
>
  Default
</div>
`;

exports[`Badge > Snapshots > matches dot indicator snapshot 1`] = `
<div
  class="inline-flex items-center rounded-md border font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-primary text-primary-foreground shadow hover:bg-primary/80 px-2.5 py-0.5 text-xs pl-1.5"
>
  <span
    aria-hidden="true"
    class="mr-1.5 h-1.5 w-1.5 rounded-full bg-current opacity-75"
  />
  With Dot
</div>
`;

exports[`Badge > Snapshots > matches removable state snapshot 1`] = `
<div
  class="inline-flex items-center rounded-md border font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-primary text-primary-foreground shadow hover:bg-primary/80 px-2.5 py-0.5 text-xs"
>
  Removable
  <button
    aria-label="Remove badge"
    class="ml-1.5 -mr-1 h-3.5 w-3.5 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
    type="button"
  >
    <svg
      class="h-3 w-3"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 18L18 6M6 6l12 12"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </button>
</div>
`;
